**Import File Data CSV**

In [ ]:
import pandas as pd

In [ ]:
file = 'https://raw.githubusercontent.com/taudata-indonesia/eLearning/master/data/price.csv'
price = pd.read_csv(file, low_memory = False, encoding = 'utf8')

N, P = price.shape #Ukuran Data
print('baris =', N, ', Kolom (jumlah variabel) =', P)
print('Tipe Variabel df =', type(price))
price

baris = 936 , Kolom (jumlah variabel) = 10
Tipe Variabel df = <class 'pandas.core.frame.DataFrame'>


Observation  Dist_Taxi  Dist_Market  Dist_Hospital  Carpet  Builtup  \
0              1     9796.0       5250.0        10703.0  1659.0   1961.0   
1              2     8294.0       8186.0        12694.0  1461.0   1752.0   
2              3    11001.0      14399.0        16991.0  1340.0   1609.0   
3              4     8301.0      11188.0        12289.0  1451.0   1748.0   
4              5    10510.0      12629.0        13921.0  1770.0   2111.0   
..           ...        ...          ...            ...     ...      ...   
931          932     9297.0      12537.0        14418.0  1174.0   1429.0   
932          933    10915.0      17486.0        15964.0  1549.0   1851.0   
933          934     9205.0      10418.0        14496.0  1118.0   1337.0   
934          935    10915.0      17486.0        15964.0  1549.0   1851.0   
935          936    10915.0      17486.0        15964.0  1549.0   1851.0   

          Parking City_Category  Rainfall  House_Price  
0            Open         CAT B       530      6649000  
1    Not Provided         CAT B       210      3982000  
2    Not Provided         CAT A       720      5401000  
3         Covered         CAT B       620      5373000  
4    Not Provided         CAT B       450      4662000  
..            ...           ...       ...          ...  
931       Covered         CAT C      1110      5434000  
932  Not Provided         CAT C      1220      7062000  
933          Open         CAT A       560      7227000  
934  Not Provided         CAT C      1220      7062000  
935  Not Provided         CAT C      1220      7062000  

[936 rows x 10 columns]

**Import Data Web API**

In [ ]:
url = 'https://en.wikipedia.org/wiki/The_World%27s_Billionaires'
df_list = pd.read_html(url) #berbentuk list
print(len(df_list))
df_list

46


[                                                    0  \
 0   List of the world's billionaires, ranked in or...   
 1   The net worth of the world's billionaires incr...   
 2                                 Publication details   
 3                                           Publisher   
 4                                         Publication   
 5                                     First published   
 6                                  Latest publication   
 7                      Current list details (2022)[2]   
 8                                          Wealthiest   
 9                                     Net worth (1st)   
 10                             Number of billionaires   
 11                         Total list net worth value   
 12                            New members to the list   
 13           Forbes: The World's Billionaires website   
 
                                                     1  
 0   List of the world's billionaires, ranked in or...  
 1   The net w

In [ ]:
df_list[0]

0  \
0   List of the world's billionaires, ranked in or...   
1   The net worth of the world's billionaires incr...   
2                                 Publication details   
3                                           Publisher   
4                                         Publication   
5                                     First published   
6                                  Latest publication   
7                      Current list details (2022)[2]   
8                                          Wealthiest   
9                                     Net worth (1st)   
10                             Number of billionaires   
11                         Total list net worth value   
12                            New members to the list   
13           Forbes: The World's Billionaires website   

                                                    1  
0   List of the world's billionaires, ranked in or...  
1   The net worth of the world's billionaires incr...  
2                                 Publication details  
3                Whale Media InvestmentsForbes family  
4                                              Forbes  
5                                       March 1987[1]  
6                                       April 5, 2022  
7                      Current list details (2022)[2]  
8                                           Elon Musk  
9                                      US$219 billion  
10                                               2668  
11                                   US$12.7 trillion  
12                                                236  
13           Forbes: The World's Billionaires website

In [ ]:
df = df_list[2]
df.head()

No.                      Name Net worth (USD)  Age    Nationality  \
0    1                 Elon Musk    $219 billion   50  United States   
1    2                Jeff Bezos    $177 billion   58  United States   
2    3  Bernard Arnault & family    $158 billion   73         France   
3    4                Bill Gates    $129 billion   66  United States   
4    5            Warren Buffett    $118 billion   91  United States   

  Source(s) of wealth  
0       Tesla, SpaceX  
1              Amazon  
2                LVMH  
3           Microsoft  
4  Berkshire Hathaway

In [ ]:
# Memilih Tabel tertentu
pd.read_html(url, match='Number and combined net worth of billionaires by year')[0].head()

Year Number of billionaires Group's combined net worth
0  2021[9]                   2755             $13.1 trillion
1     2020                   2095              $8.0 trillion
2     2019                   2153              $8.7 trillion
3     2018                   2208              $9.1 trillion
4     2017                   2043              $7.7 trillion

# **Contoh Kasus Data Harga Property Rumah**

In [ ]:
import warnings; warnings.simplefilter('ignore')
import scipy, itertools, pandas as pd, matplotlib.pyplot as plt, seaborn as sns, numpy as np
from scipy import stats
from sklearn.preprocessing import StandardScaler, MinMaxScaler

plt.style.use('bmh'); sns.set()

In [ ]:
price.head(9)

Observation  Dist_Taxi  Dist_Market  Dist_Hospital  Carpet  Builtup  \
0            1     9796.0       5250.0        10703.0  1659.0   1961.0   
1            2     8294.0       8186.0        12694.0  1461.0   1752.0   
2            3    11001.0      14399.0        16991.0  1340.0   1609.0   
3            4     8301.0      11188.0        12289.0  1451.0   1748.0   
4            5    10510.0      12629.0        13921.0  1770.0   2111.0   
5            6     6665.0       5142.0         9972.0  1442.0   1733.0   
6            7    13153.0      11869.0        17811.0  1542.0   1858.0   
7            8     5882.0       9948.0        13315.0  1261.0   1507.0   
8            9     7495.0      11589.0        13370.0  1090.0   1321.0   

        Parking City_Category  Rainfall  House_Price  
0          Open         CAT B       530      6649000  
1  Not Provided         CAT B       210      3982000  
2  Not Provided         CAT A       720      5401000  
3       Covered         CAT B       620      5373000  
4  Not Provided         CAT B       450      4662000  
5          Open         CAT B       760      4526000  
6    No Parking         CAT A      1030      7224000  
7          Open         CAT C      1020      3772000  
8  Not Provided         CAT B       680      4631000

In [ ]:
price.tail(4)

Observation  Dist_Taxi  Dist_Market  Dist_Hospital  Carpet  Builtup  \
932          933    10915.0      17486.0        15964.0  1549.0   1851.0   
933          934     9205.0      10418.0        14496.0  1118.0   1337.0   
934          935    10915.0      17486.0        15964.0  1549.0   1851.0   
935          936    10915.0      17486.0        15964.0  1549.0   1851.0   

          Parking City_Category  Rainfall  House_Price  
932  Not Provided         CAT C      1220      7062000  
933          Open         CAT A       560      7227000  
934  Not Provided         CAT C      1220      7062000  
935  Not Provided         CAT C      1220      7062000

In [ ]:
#pilih random
price.sample(10)

Observation  Dist_Taxi  Dist_Market  Dist_Hospital  Carpet  Builtup  \
543          544     7669.0      10444.0        12941.0  1520.0   1817.0   
224          225     9024.0       9404.0        13234.0  1863.0   2245.0   
331          332     3114.0       7163.0         8477.0  1375.0   1659.0   
195          196    10813.0      12852.0        16833.0  1524.0   1796.0   
253          254     4906.0      10462.0        12246.0  1539.0   1848.0   
546          547     9089.0       9503.0        13179.0  1777.0   2117.0   
155          156     7738.0       9475.0        14329.0  1229.0   1480.0   
417          418    11445.0       5955.0        13648.0  2063.0   2493.0   
205          206    10502.0      12011.0        16938.0  1841.0   2215.0   
542          543     6493.0      14696.0        14315.0  1351.0   1616.0   

          Parking City_Category  Rainfall  House_Price  
543          Open         CAT A       850      6360000  
224          Open         CAT A       640      7726000  
331       Covered         CAT A      1080      7664000  
195          Open         CAT C      1060      4108000  
253          Open         CAT B       750      4714000  
546  Not Provided         CAT B       780      3828000  
155          Open         CAT C       830      3807000  
417       Covered         CAT C       810      5148000  
205       Covered         CAT A       610      8876000  
542       Covered         CAT B       650      6277000

**loc vs. iloc**

In [ ]:
df_train = price.sample(300)
df_train.head()

Observation  Dist_Taxi  Dist_Market  Dist_Hospital  Carpet  Builtup  \
484          485     7971.0       8165.0        12202.0  1845.0   2233.0   
233          234    13071.0      15127.0        17387.0  1251.0   1527.0   
400          401     9154.0      12441.0        14070.0  1208.0   1467.0   
420          421    11309.0       8252.0        13199.0  1480.0   1773.0   
461          462     7136.0       8734.0         9830.0  1721.0   2076.0   

          Parking City_Category  Rainfall  House_Price  
484    No Parking         CAT B       370      4319000  
233          Open         CAT C       380      6383000  
400  Not Provided         CAT B       910      5072000  
420       Covered         CAT B       360      6302000  
461    No Parking         CAT A       710      7630000

**loc** -> mengambil data berdasarkan nilai index, seperti pada kasus di atas adalah 484, 233, 400, 420, dan 461.

**iloc** -> mengambil data berdasarkan urutan index, seperti pada kasus di atas adalah 484 urutan index 0, 233 urutan index 1, 200 urutan index 2, dst.

In [ ]:
print(df_train.loc[484,'Parking'])
df_train.loc[484]

No Parking


Observation             485
Dist_Taxi            7971.0
Dist_Market          8165.0
Dist_Hospital       12202.0
Carpet               1845.0
Builtup              2233.0
Parking          No Parking
City_Category         CAT B
Rainfall                370
House_Price         4319000
Name: 484, dtype: object

In [ ]:
print(df_train.iloc[0]['Parking'])
df_train.iloc[0]

No Parking


Observation             485
Dist_Taxi            7971.0
Dist_Market          8165.0
Dist_Hospital       12202.0
Carpet               1845.0
Builtup              2233.0
Parking          No Parking
City_Category         CAT B
Rainfall                370
House_Price         4319000
Name: 484, dtype: object

**Removing a varible**

In [ ]:
price.columns

Index(['Observation', 'Dist_Taxi', 'Dist_Market', 'Dist_Hospital', 'Carpet',
       'Builtup', 'Parking', 'City_Category', 'Rainfall', 'House_Price'],
      dtype='object')

In [ ]:
price.drop('Observation', axis=1, inplace=True)
price.head()

Dist_Taxi  Dist_Market  Dist_Hospital  Carpet  Builtup       Parking  \
0     9796.0       5250.0        10703.0  1659.0   1961.0          Open   
1     8294.0       8186.0        12694.0  1461.0   1752.0  Not Provided   
2    11001.0      14399.0        16991.0  1340.0   1609.0  Not Provided   
3     8301.0      11188.0        12289.0  1451.0   1748.0       Covered   
4    10510.0      12629.0        13921.0  1770.0   2111.0  Not Provided   

  City_Category  Rainfall  House_Price  
0         CAT B       530      6649000  
1         CAT B       210      3982000  
2         CAT A       720      5401000  
3         CAT B       620      5373000  
4         CAT B       450      4662000

**Mengoreksi Tipe Variable**

In [ ]:
price.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 936 entries, 0 to 935
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Dist_Taxi      923 non-null    float64
 1   Dist_Market    923 non-null    float64
 2   Dist_Hospital  935 non-null    float64
 3   Carpet         928 non-null    float64
 4   Builtup        921 non-null    float64
 5   Parking        936 non-null    object 
 6   City_Category  936 non-null    object 
 7   Rainfall       936 non-null    int64  
 8   House_Price    936 non-null    int64  
dtypes: float64(5), int64(2), object(2)
memory usage: 65.9+ KB


In [ ]:
price.dtypes

Dist_Taxi        float64
Dist_Market      float64
Dist_Hospital    float64
Carpet           float64
Builtup          float64
Parking           object
City_Category     object
Rainfall           int64
House_Price        int64
dtype: object

In [ ]:
price['Parking'] = price['Parking'].astype('category')
price['City_Category'] = price['City_Category'].astype('category')
price.dtypes

Dist_Taxi         float64
Dist_Market       float64
Dist_Hospital     float64
Carpet            float64
Builtup           float64
Parking          category
City_Category    category
Rainfall            int64
House_Price         int64
dtype: object